File: airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/MySqlSourceExceptionHandlerTest.java
/*
 * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.integrations.source.mysql;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.io.EOFException;
import java.sql.SQLSyntaxErrorException;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

public class MySqlSourceExceptionHandlerTest {

  private MySqlSourceExceptionHandler exceptionHandler;

  @BeforeEach
  void setUp() {
    exceptionHandler = new MySqlSourceExceptionHandler();
    exceptionHandler.initializeErrorDictionary();
  }

  @Test
  void testTranslateMySQLSyntaxException() {
    SQLSyntaxErrorException exception = new SQLSyntaxErrorException("Unknown column 'xmin' in 'field list'");
    String external_message = exceptionHandler.getExternalMessage(exception);
    assertEquals("A column needed by MySQL source connector is missing in the database", external_message);
  }

  @Test
  void testTranslateEOFException() {
    EOFException exception =
        new EOFException("Can not read response from server. Expected to read 4 bytes, read 0 bytes before connection was unexpectedly lost.");
    String external_message = exceptionHandler.getExternalMessage(exception);
    assertEquals("Can not read data from MySQL server", external_message);
  }

}

File: airbyte-integrations/connectors/source-mysql/src/test/kotlin/MySqlSourceExceptionHandlerTest.kt
/*
 * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
 */

import io.airbyte.cdk.integrations.util.FailureType
import io.airbyte.integrations.source.mysql.MySqlSourceExceptionHandler
import java.io.EOFException
import java.sql.SQLSyntaxErrorException
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class MySqlSourceExceptionHandlerTest {
    private var exceptionHandler: MySqlSourceExceptionHandler? = null

    @BeforeEach
    fun setUp() {
        exceptionHandler = MySqlSourceExceptionHandler()
    }

    @Test
    fun testTranslateMySQLSyntaxException() {
        val exception = SQLSyntaxErrorException("Unknown column 'xmin' in 'field list'")
        val externalMessage = exceptionHandler!!.getExternalMessage(exception)
        Assertions.assertTrue(exceptionHandler!!.checkErrorType(exception, FailureType.CONFIG))
        Assertions.assertEquals(
            "A column needed by MySQL source connector is missing in the database",
            externalMessage
        )
    }

    @Test
    fun testTranslateEOFException() {
        val exception =
            EOFException(
                "Can not read response from server. Expected to read 4 bytes, read 0 bytes before connection was unexpectedly lost."
            )
        val externalMessage = exceptionHandler!!.getExternalMessage(exception)
        Assertions.assertTrue(exceptionHandler!!.checkErrorType(exception, FailureType.TRANSIENT))
        Assertions.assertEquals("Can not read data from MySQL server", externalMessage)
    }
}
